exporter/prometheus.go

```diff
@@ -46,3 +66,55 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	log.Info("All Metrics successfully collected")
 
 }
+
+func (e *Exporter) isTokenExpired() (bool, error) {
+	u := *e.APIURL()
+	u.Path = path.Join(u.Path, "rate_limit")
+
+	resp, err := getHTTPResponse(u.String(), e.APIToken())
+
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+	// Triggers if rate-limiting isn't enabled on private Github Enterprise installations
+	if resp.StatusCode == 404 {
+		return false, errors.New("404 Error")
```

Comment:
IMPORTANT: i'm not too familiar with GH Enterprise but if rate-limiting isn't enabled, then shouldn't this be returning `false, nil`? otherwise it will return from the calling function before it has collected metrics

Reply:
Correct, Done.